File: expenses-manager-backend-java/src/main/java/com/github/beliakou/expenses/repository/UserRepository.java
package com.github.beliakou.expenses.repository;

import com.github.beliakou.expenses.domain.User;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public class UserRepository {

    private MongoTemplate mongoTemplate;

    public UserRepository(MongoTemplate mongoTemplate) {
        this.mongoTemplate = mongoTemplate;
    }

    public Optional<User> findUser(String name) {
        return Optional.ofNullable(this.mongoTemplate.findOne(
                Query.query(Criteria.where("name").is(name)), User.class, "user"));
    }
}

File: expenses-manager-backend/expenses-manager-server/src/main/kotlin/com/github/beliakou/expenses/server/repository/UserRepository.kt
package com.github.beliakou.expenses.server.repository

import com.github.beliakou.expenses.server.model.User
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource

@RepositoryRestResource(collectionResourceRel = "users", path = "users")
interface UserRepository : MongoRepository<User, String> {

    fun findByName(name: String): User
}